### PR TITLE
fix nget handling, provide ngetscu example

### LINKED ISSUE #77: Fix/add N-GET support
While N-GET is not required for TDW-II it is available as one of the UPS primitives.
The current implementation is unlikely to work correctly based on it's use of the pynetdicom event.identifier property that throws an exception if the (pynetdicom, not UPS) event is not C-FIND or C-GET or C-MOVE.

### PR DESCRIPTION
**Pull request by sjswerdloff**
avoid use of event identifier, use RequestedSOPClassUID/RequestedSOPInstanceUID for model and match criteria.
This doesn't narrow down the N-GET-RSP to just those elements requested.
A separate feature request should be created to accomplish that.
Should close #77

### CODE REVIEW

**Comment by sjswerdloff** [comment]
@christof-schadt-bl and @dwikler sorry, I didn't remember who had brought up the N-GET (which isn't a part of TDW-II but is a part of UPS... so that's part of the "plus")

File: tdwii_plus_examples/handlers.py
Comment (dwikler) [inline tdwii_plus_examples/handlers.py:1]: at line 378, I wonder how ds_from_request initialized in this method can contain the identifier. I do not see the event.identifier has been added to ds_from_request

File: tdwii_plus_examples/handlers.py [hunk 1]
```diff
@@ -357,18 +357,28 @@ def handle_nget(event, db_path, cli_config, logger):
...
+        print("Requested SOP Instance UID is empty")
 
     engine = create_engine(db_path)
     with engine.connect() as conn:  # noqa:  F841
         Session = sessionmaker(bind=engine)
         session = Session()
         # Search database using Identifier as the query
         try:
-            matches = search(model, event.identifier, session)
+            matches = search(model, ds_from_request, session)
```
Comment (dwikler) [inline tdwii_plus_examples/handlers.py:378]: at line 378, I wonder how ds_from_request initialized in this method can contain the identifier. I do not see the event.identifier has been added to ds_from_request

File: tdwii_plus_examples/ngetscu.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import tdwii_plus_examples.tdwii_config as ae_config
+
+
+def nget_ups(
+    attribute_list: List[DataElement], ups_instance_uid: str, scu_ae_title: str, scp_ae_title: str
+) -> Iterator[tuple[Dataset, Dataset | None]]:
+    """Perform UPS N-GET-RQ and return responses that have UPS content
+
+    Args:
+        ds_query (pydicom.Dataset): query content/request
```
Comment (dwikler) [inline tdwii_plus_examples/ngetscu.py:20]: ds_query not in the args. attribute_list and ups_instance_uid missing from this docstring

File: tdwii_plus_examples/ngetscu.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    if scp_ae_title not in ae_config.known_ae_ipaddr:
+        logging.error("Unable to query " + scp_ae_title + " not found in this TDW-II Application Entity configuration")
+        return
+    identifier_list = set([x.tag for x in attribute_list])
+    # SOP Class UID, SOP Instance UID, Transaction UID are not allowed
+    disallowed_tags = set([0x00080016, 0x00080018, 0x00081195])
+    identifier_list.difference_update(disallowed_tags)
+    for tag in identifier_list:
+        print(tag)
```
Comment (dwikler) [inline tdwii_plus_examples/ngetscu.py:40]: may be modified to logging.info

**Review by dwikler — COMMENTED** [review]
I have a doubt about how the identifier is conveyed in the search, see comment in handlers.py

File: tdwii_plus_examples/handlers.py (hunk 1 above)
Comment (sjswerdloff) [reply tdwii_plus_examples/handlers.py:378]: event.identifier is not populated by pynetdicom for N-GET.  Which is why it was failing.  
A few lines higher up is
`ds_from_request.SOPInstanceUID = event.request.RequestedSOPInstanceUID`
is how the required elements to match is provided (the Requested SOP Instance UID in the N-GET-RQ is used to look up the UPS SOP Instance UID)

File: tdwii_plus_examples/ngetscu.py (hunk 2 above)
Comment (sjswerdloff) [reply tdwii_plus_examples/ngetscu.py]: fixed in 0e0d99882130b7aee1bd5f0ae0b6df0193e7b4e2

File: tdwii_plus_examples/ngetscu.py (hunk 3 above)
Comment (sjswerdloff) [reply tdwii_plus_examples/ngetscu.py]: fixed in 0e0d99882130b7aee1bd5f0ae0b6df0193e7b4e2

**Review by dwikler — APPROVED** [review]

**Pull request merged**